src/Interpreters/Cache/FileCache.cpp

```diff
@@ -1349,6 +1354,49 @@ void FileCache::assertCacheCorrectness()
     });
 }
 
+void FileCache::applySettingsIfPossible(const FileCacheSettings & new_settings, FileCacheSettings & actual_settings)
```

Comment:
Carrying around `actual_settings` is not necessary if applying `new_settings` as a whole.

Reply:
Answered in https://github.com/ClickHouse/ClickHouse/pull/57578#discussion_r1421404672.